pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
The idea here is that we'd diff the original config (stored in an annotation) and updated config, and then apply the patch to the live state?

That's a reasonable starting point.

We'll also want to be able to detect where the live state has diverged from the original (and perhaps new) configuration.